js/languagePickerModel.js

```diff
@@ -11,8 +11,18 @@ define([
             "_languages": []
         },
 
+        trackedData: {
+            "components": [],
+            "blocks": []
+        },
+
+        locationId: null,
+
         initialize: function () {
             this.listenTo(Adapt.config, 'change:_activeLanguage', this.onConfigChange);
+            if (this.get('_restoreStateOnLanguageChange')) {
```

Comment:
you can get rid of this check now i.e. always listen to the `app:dataLoaded` event